test/e2e/downwardapi_volume.go

```diff
@@ -81,11 +83,30 @@ var _ = Describe("Downward API volume", func() {
 		},
 			podLogTimeout, poll).Should(ContainSubstring("key1=\"value1\"\n"))
 
-		//modify labels
-		pod.Labels["key3"] = "value3"
-		_, err = f.Client.Pods(f.Namespace.Name).Update(pod)
-		Expect(err).NotTo(HaveOccurred())
-
+		//modify labels	in a update/get loop
+		expectNoError(wait.Poll(time.Millisecond*500, time.Second*30, func() (bool, error) {
+			By("updating the pod")
+			if pod == nil { // on retries we need to re-get
+				pod, err = f.Client.Pods(f.Namespace.Name).Get(pod.Name)
+				if err != nil {
+					return false, fmt.Errorf("failed to get pod: %v", err)
+				}
+			}
+			pod.Labels["key3"] = "value3"
+			pod, err = f.Client.Pods(f.Namespace.Name).Update(pod)
+			if err == nil {
+				Logf("Successfully updated pod")
+				return true, nil
+			}
+			if errors.IsConflict(err) {
```

Comment:
i think if we don't do it now we will probably eventually add a loop for retries -- want to make it a loop w/ 2 retries for now?